### PR TITLE
kata-deploy: Add support for qemu and kernel with virtio-fs 3.0

### LINKED ISSUE #709: Add kata deploy for QEMU  and kernel with virtio-fs 3.0

### PR DESCRIPTION
**Pull request by GabyCT**
This adds the kata deploy for QEMU and kernel with virtio-fs 3.0

Depends-on: github.com/kata-containers/runtime#2052

Fixes #709

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: kata-deploy/examples/test-deploy-kata-qemu-virtiofs.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          protocol: TCP
+        resources:
+          requests:
+            cpu: 200m
+      restartPolicy: Always
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: php-apache-kata-qemu-virtiofs
```
Comment (jcvenegas) [inline kata-deploy/examples/test-deploy-kata-qemu-virtiofs.yaml]: We dont have a test today for this you better try locally to see if we are not missing something here :)

**Comment by nitkon** [comment]
@GabyCT :  Are arch specific kernel fragment directory a must to build experimental kernel for virtio-fs?
```
++ kernel_ver=virtio-fs-v0.3
++ archfragdir=/home/nitesh/split_patch/src/github.com/kata-containers/packaging/kernel/configs/fragments/powerpc
++ '[' -d /home/nitesh/split_patch/src/github.com/kata-containers/packaging/kernel/configs/fragments/powerpc ']'
++ config=/home/nitesh/split_patch/src/github.com/kata-containers/packaging/kernel/configs/powerpc_kata_kvm_virtio-fs-v0.3.x
++ '[' -f /home/nitesh/split_patch/src/github.com/kata-containers/packaging/kernel/configs/powerpc_kata_kvm_virtio-fs-v0.3.x ']'
++ die 'failed to find default config /home/nitesh/split_patch/src/github.com/kata-containers/packaging/kernel/configs/powerpc_kata_kvm_virtio-fs-v0.3.x'
++ echo 'ERROR: failed to find default config /home/nitesh/split_patch/src/github.com/kata-containers/packaging/kernel/configs/powerpc_kata_kvm_virtio-fs-v0.3.x'
ERROR: failed to find default config /home/nitesh/split_patch/src/github.com/kata-containers/packaging/kernel/configs/powerpc_kata_kvm_virtio-fs-v0.3.x
++ exit 1
+ kernel_config_path=
```

**Comment by GabyCT** [comment]
@nitkon probably @egernst is the person who can answer that as he took that task

**Comment by GabyCT** [comment]
@jcvenegas changes applied, here are the related PRs
https://github.com/kata-containers/runtime/pull/2052
https://github.com/kata-containers/runtime/pull/2054

File: kernel/kata_config_version [hunk 2]
```diff
@@ -1 +1 @@
-49
+50
```
Comment (nitkon) [inline kernel/kata_config_version]: I see you have bumped the version here but still CI is failing with: 
```
23:01:19 Kernel directory has changes check kernel/kata_config_version changed
23:01:19 ERROR: Please bump version in kernel/kata_config_version
23:01:19 Makefile:24: recipe for target 'test' failed
```

Comment (GabyCT) [reply kernel/kata_config_version]: @jcvenegas any idea of this failure?

Comment (GabyCT) [reply kernel/kata_config_version]: @jcvenegas and @nitkon, I just fixed it

File: kernel/build-kernel.sh [hunk 3]
```diff
@@ -414,9 +416,13 @@ main() {
 
 	# If not kernel version take it from versions.yaml
 	if [ -z "$kernel_version" ]; then
-		kernel_version=$(get_from_kata_deps "assets.kernel.version")
-		#Remove extra 'v'
-		kernel_version="${kernel_version#v}"
+		if [[ ${experimental_kernel} == "true" ]]; then
+			kernel_version="virtio-fs-v0.3"
```
[the remark this answers is not in the document]
Comment (egernst) [reply kernel/build-kernel.sh]: Yes, ultimately this should end up coming from versions.yaml.

File: kata-deploy/scripts/kata-deploy-docker.sh [hunk 4]
```diff
@@ -40,6 +40,10 @@ function configure_docker() {
       "path": "/opt/kata/bin/kata-runtime",
       "runtimeArgs": [ "--kata-config", "/opt/kata/share/defaults/kata-containers/configuration-qemu.toml" ]
     },
+    "kata-qemu-virtiofs": {
+      "path": "/opt/kata/bin/kata-runtime",
+      "runtimeArgs": [ "--kata-config", "/opt/kata/share/defaults/kata-containers/configuration-qemu-virtiofs.toml" ]
```
[the remark this answers is not in the document]
Comment (egernst) [reply kata-deploy/scripts/kata-deploy-docker.sh]: @GabyCT we should probably go ahead and add a virtio-fs specific toml to the runtime too.

File: kernel/build-kernel.sh [hunk 5]
```diff
@@ -78,10 +78,12 @@ Commands:
 
 Options:
 
-	-c <path>: Path to config file to build a the kernel
+	-c <path>: Path to config file to build a the kernel.
+	-e       : Enable experimental kernel.
```
[the remark this answers is not in the document]
Comment (egernst) [reply kernel/build-kernel.sh]: Thanks, mea culpa!

File: release/kata-deploy-binaries.sh [hunk 6]
```diff
@@ -103,6 +103,15 @@ install_kernel() {
 	)
 }
 
+#Install experimental kernel asset
+install_experimental_kernel() {
+	info "build experimental kernel"
```
[the remark this answers is not in the document]
Comment (amshinde) [reply release/kata-deploy-binaries.sh]: @GabyCT I suppose #722 addresses the above issue that @jcvenegas pointed out?
If so, does that PR need to go before this one?

**Comment by GabyCT** [comment]
> Looks good. I wonder if we should wait for a runtime toml to be in place before merging?

@egernst , I already submitted the changes for the runtime toml https://github.com/kata-containers/runtime/pull/2054

**Comment by GabyCT** [comment]
@amshinde , I include the changes at https://github.com/kata-containers/runtime/pull/2054

**Pull request merged**